instancio-quickcheck/src/main/java/org/instancio/quickcheck/internal/engine/InstancioQuickcheckEngineDescriptor.java

```diff
@@ -13,20 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.instancio.quickcheck.engine;
+package org.instancio.quickcheck.internal.engine;
```

Comment:
Moving engine-related classes under the `internal` package, assuming they are not part of the public API.

Reply:
This is a bit questionable (in my opinion) - we expose engine through service loader mechanism so whatever the strategy is used, this class has to be publicly available (or in case of JPMS, be part of `module-info`), this is was a motivation for me to not have it under `internal` package, what do you think?

Reply:
I think we can always expose the `InstancioQuickcheckTestEngine` in the future if users request it. For now, I added a quickcheck [JPMS test](https://github.com/instancio/instancio-jpms-tests) and it seems to work. The only issue is that it requires adding `add-exports` because we're currently using a few of JUnit's internal packages:

```
--add-exports org.junit.jupiter.engine/org.junit.jupiter.engine.descriptor=org.instancio.quickcheck
--add-exports org.junit.jupiter.engine/org.junit.jupiter.engine.discovery.predicates=org.instancio.quickcheck
--add-exports org.junit.platform.commons/org.junit.platform.commons.logging=org.instancio.quickcheck
--add-exports org.junit.platform.commons/org.junit.platform.commons.util=org.instancio.quickcheck
```

We can refactor the code later so that users aren't forced to add these. E.g.

- `org.junit.platform.commons.util.ReflectionUtils` can be replaced with `org.junit.platform.commons.support.ReflectionSupport`
- `AnnotationUtils` with `AnnotationSupport`
- JUnit's logging can be replaced with `slf4j-api`.